broadinstitute/wfl
# [GH-886] Have OAuth2 Client ID be handled by the backend

### Purpose


- https://broadinstitute.atlassian.net/browse/GH-886

### Changes


- Add an unauthenticated endpoint to WFL to return the OAuth2 Client ID correlating with WFL's environment (getting it from environment if possible, then from Vault, then just the dev one for backwards compatibility)
- Add a test for said endpoint
- Modify UI to call said endpoint to get the client ID before running the OAuth2 sequence
- Add references to the environment variable and endpoint to the Helm config

### Review Instructions


- Run integration tests
- Run a local server and see that you can still log in--this verifies that WFL's endpoint returned the OAuth2 Client ID it got from vault and that the login sequence still works
- Go to https://jwarren-wfl.gotc-dev.broadinstitute.org/oauth2id to verify that the endpoint works unauthenticated on a real deployment (that's my own sandbox running off of the exact same wfl-instance and deployment setup that dev and prod are on)
    - If you try to log in on my sandbox, it will fail _intentionally_: the old default of the dev ID (still used as a fallback) actually works on my sandbox, so for this deployment I manually edited the values file to use the prod ID. The fact that it fails with a "hey! you're on the wrong URL to be using this ID!" error is super important and actually means the whole thing is working, because it means that the prod ID got used during the log in process like I asked. If you want to see the log in process work, just run a local server or let me know.
